workflow.md

```diff
@@ -100,10 +123,91 @@ Please note that actual version of project is defined by project settings,
-as well.
+as well (corresponding to the latest release).
+
+### Special workflow for cardano-sl project
+
+Table below illustrates a special workflow used for cardano-sl project
+with respect to branches and deployment clusters.
+
+| Branch             | Deployment  | Purpose                                        |
+| ------------------ | ----------- | ---------------------------------------------- |
+| master             | csl-testing | Testing current master (by devs)               |
+| cardano-sl-X.Y-rc  | csl-qa      | QA testing of public release candidate (qanet) |
+| cardano-sl-X.Y     | csl-testnet | Public release (testnet)                       |
+| cardano-sl-X.Y-mid | ——————————— | Used by other projects (e. g. explorer)        |
+
+Flow should be following:
+1. When version is being wrapped up, `master` should be deployed to
+   `csl-testing` cluster, problems arising right away should be
+   resolved.
+2. After successful deployment on `csl-testing`:
+    a) `vX.Y.0` tag should be set (in `master`).
+    b) `cardano-sl-X.Y-rc` should be branched out of `master`.
+    c) `cardano-sl-X.Y-rc` should be deployed on `csl-qa` cluster.
+    d) All bugs discovered should be fixed in `-rc` branch, version
+	is increased for each fix.
+3. After `X.Y` version is polished and is ready to be published (it's
+`vX.Y.Z` already):
+    a) `cardano-sl-X.Y-rc` should be renamed to `cardano-sl-X.Y`, `cardano-sl-X.Y-mid` immediately spawned
+    b) `cardano-sl-X.Y-mid` is immediately spawned from the same commit.
+    c) `genesis.bin`, `constants-*.yaml` should be changed.
+    d) `vX.Y.Z` should be deployed on `csl-testnet` cluster.
+    e) tada.iohk.io, daedaluswallet.io should be updated.
+
+Rules for `-rc` branch are same as for release branch, specifically:
+* `-rc` branch must be merged into `master` after every change.
+* `master` is never merged into `-rc` branch.
+* `-rc` branch must be protected.
+
+### Interaction between cardano-sl and projects dependent on it
+
+Some projects (e. g. `cardano-sl-explorer`, `daedalus`) depend on
+`cardano-sl`. In order to facilitate interaction with `cardano-sl` we
+use special branch `-mid` mentioned above. Explorer is used as an
+example in the rest of this subsection, but rules apply to everything
+dependent on `cardano-sl`.
+
+* Explorer must depend only on version of `cardano-sl` from release
+  branch or `-mid` branch, not `-rc` branch. This rule is imposed
+  because only these branches are generally stable and
+  well-tested. `-rc` branch corresponds to a version which is not
+  properly tested yet, hence explorer should not depend on it.
+* Explorer should always be launched with the same version of
+  `cardano-sl` as the version it depends on. That's because different
+  versions of `cardano-sl` may be incompatible in some way and it may
+  lead to different (inconsistent) behavior of `cardano-node` and
+  `cardano-explorer`.
+* Version `X.Y.Z` of explorer (located in `sl-explorer-X.Y` branch)
```

Comment:
@jagajaga @gromakovsky 

Here is my proposal to update chapter `Interaction between cardano-sl and projects dependent on it`

```
### Interaction between cardano-sl and projects dependent on it

Some projects (e. g. `cardano-sl-explorer`, `daedalus`) depend on
`cardano-sl`. In order to facilitate interaction with `cardano-sl` we
use special branch `-mid` mentioned above. `Explorer` is used as an
example in the rest of this subsection, but rules apply to everything
dependent on `cardano-sl`.

* `Explorer` must depend on version of `cardano-sl` from `-mid` branch only,
  not `-rc` or release branch. This rule is imposed
  because only this branch is generally stable and
  well-tested for using it by third party applications.
  `-rc` branch corresponds to a version which is not
  properly tested yet, hence explorer should not depend on it.
* `Explorer` comes with its own version based on [CSL versioning](Versions).
  However, to reflect current version of `cardano-sl`, Explorer is using a branch
  `cardano-sl-X.Y`, where `X.Y` is `Maj.Min` of `cardano-sl-X.Y-mid`.
  That's because different versions of `cardano-sl` may be incompatible in some way and it may
  lead to different (inconsistent) behavior of `cardano-node` and
  `cardano-explorer`. For example: `Explorer's` branch `cardano-sl-0.4` is pointing to `0.4` of `cardano-sl`.

* If `Explorer` developer wants to change something in `cardano-sl` for
  `Explorer's` version `X.Y.Z` they should make a PR to `cardano-sl-X.Y`
  branch, wait until it's merged and then use newer version in
  explorer.
* `master` branch of `Explorer` should depend on the latest `-mid`
  branch of `cardano-sl`.
* If `Explorer` developers wants to change something in `cardano-sl` for
  explorer's `master` they should make a PR to `cardano-sl-X.Y-mid` of
  `cardano-sl` (where `X.Y` is latest version of `cardano-sl`).
* `cardano-sl-X.Y.` should be frequently merged into
  `cardano-sl-X.Y-mid`.
* `cardano-sl-X.Y-mid` should be frequently merged to `master`.
```